### PR TITLE
Remove redundant calls to `borrow()`

### PR DESCRIPTION
**Pull request by mrcnski**
Fixes a few of these warnings I get with a recent nightly:

```
warning: call to `.borrow()` on a reference in this situation does nothing
   --> substrate/frame/support/src/storage/generator/map.rs:317:60
    |
317 |                 Some(ref val) => unhashed::put(final_key.as_ref(), &val.borrow()),
    |                                                                        ^^^^^^^^^ help: remove this redundant call
    |
    = note: the type `V` does not implement `Borrow`, so calling `borrow` on `&V` copies the reference, which does not do anything and can be removed

```

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3557265

**Review by juangirini — COMMENTED** [review]
I wonder why it was there in the first place...

**Review by juangirini — APPROVED** [review]

**Review by ggwpez — APPROVED** [review]

**Review by sandreim — APPROVED** [review]

**Review by franciscoaguirre — APPROVED** [review]

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3559608

**Review by bkchr — APPROVED** [review]

File: substrate/client/api/src/lib.rs [hunk 1]
```diff
@@ -69,10 +69,8 @@ pub mod utils {
 				return Ok(false)
 			}
 
-			let current = current.as_ref().map(|(c, p)| (c.borrow(), p.borrow()));
-
 			let mut hash = hash;
-			if let Some((current_hash, current_parent_hash)) = current {
+			if let Some((current_hash, current_parent_hash)) = current.as_ref() {
```
Comment (KiChjang) [inline substrate/client/api/src/lib.rs:73]: This should just work too:

```suggestion
			if let Some((current_hash, current_parent_hash)) = &current {
```

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable-int
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/3564063

**Comment by mrcnski** [comment]
bot merge

**Pull request merged**